ecs-microservice/grafana.tf

```diff
@@ -10,20 +10,35 @@
 data "aws_region" "current" {}
 
 resource "grafana_folder" "collection" {
-  count = (var.grafana_use_existing_folder==-1 && var.grafana_create_dashboard == true) ? 1 : 0
-  title = length(var.grafana_folder_name)>0 ? var.grafana_folder_name : title("${var.name_prefix} > ${var.service_name}")
+  count = var.grafana_create_dashboard == true ? 1 : 0
+  title = title("${var.name_prefix} > ${var.service_name} > ${var.environment}")
 }
 
 resource "grafana_dashboard" "dashboard_in_folder" {
-  count = var.grafana_create_dashboard == true ? 1 : 0
-  folder = var.grafana_use_existing_folder>0 ? var.grafana_use_existing_folder : grafana_folder.collection[0].id
-  config_json = templatefile(length(var.grafana_template_file)>0 ? var.grafana_template_file : "${path.module}/default-dashboard.tpl", {
-    "name": title("${var.service_name} ${var.environment}")
-    "environment": var.environment
-    "name_prefix": var.name_prefix
-    "application": var.service_name
-    "service_name": var.service_name
-    "region": "eu-west-1"
-    "uuid": filemd5(length(var.grafana_template_file)>0 ? var.grafana_template_file : "${path.module}/default-dashboard.tpl")
+  count  = var.grafana_create_dashboard == true ? 1 : 0
+  folder = grafana_folder.collection[0].id
+  config_json = templatefile("${path.module}/grafana-templates/ecs-dashboard.tpl", {
+    "name" : title("ECS ${var.service_name} ${var.environment}")
+    "environment" : var.environment
+    "name_prefix" : var.name_prefix
+    "application" : var.service_name
+    "service_name" : var.service_name
+    "region" : "eu-west-1"
+    "uuid" : md5("ECS ${var.name_prefix} > ${var.service_name} > ${var.environment}")
+  })
+}
+
+resource "grafana_dashboard" "rds_dashboard_in_folder" {
+  count  = var.grafana_create_dashboard == true && length(var.rds_enabled) > 0 ? 1 : 0
+  folder = grafana_folder.collection[0].id
+  config_json = templatefile("${path.module}/grafana-templates/rds-dashboard.tpl", {
+    "name" : title("RDS ${var.service_name} ${var.environment}")
+    "environment" : var.environment
+    "name_prefix" : var.name_prefix
+    "application" : var.service_name
+    "service_name" : var.service_name
+    "db_instance_identifier": var.grafana_configuration.db_instance_identifier
```

Comment:
burde heller plukkes direkte fra paramerne til ecs-modulen, enn å innføre et eget parameter objekt for dashboardet syns jeg. hvis det mangler et parameter for rds database, så ser det ut som en mangel i parameterne til ecs-modulen, gir man kanskje tilgang for alle servicer til å å koble seg til alle rds baser ettersom det ikke finnes en spesifikk parameter med navnet til databasen for servicen?

å innføre et eget grafana dashboard parameterobjekt blir bare en patch på et symptom isåfall.

Reply:
Ja, du har rett. Gjør om på den til en helt egen variabel